system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs

```diff
@@ -404,14 +404,21 @@ impl xcm_executor::Config for XcmConfig {
 /// Forms the basis for local origins sending/executing XCMs.
 pub type LocalSignedOriginToLocation = SignedToAccountId32<RuntimeOrigin, AccountId, RelayNetwork>;
 
-/// For routing XCM messages which do not cross local consensus boundary.
-type LocalXcmRouter = (
+/// Use [`LocalXcmRouter`] instead.
+pub(crate) type LocalXcmRouterWithoutException = (
 	// Two routers - use UMP to communicate with the relay chain:
 	cumulus_primitives_utility::ParentAsUmp<ParachainSystem, PolkadotXcm, PriceForParentDelivery>,
 	// ..and XCMP to communicate with the sibling chains.
 	XcmpQueue,
 );
 
+/// For routing XCM messages which do not cross local consensus boundary.
+type LocalXcmRouter = pallet_ah_migrator::RouteInnerWithException<
+	LocalXcmRouterWithoutException,
+	Equals<ParentLocation>,
+	crate::AhMigrator,
```

Comment:
Not sure i understand it here: I thought we want to block messages from all origins that are not `ParentLocation`.  
But looking at the code of `RouteInnerWithException` it seems like it will block `Equals<ParentLocation>` when the migration is ongoing.

Reply:
we do not lock anyone from sending to RC or AH and they will use that 2 blocks from our pattern (lets say (18, 2)). but we do not let their xcm programs to send messages from RC or AH to AH or RC and use that 18 blocks from our pattern.

Reply:
@muharem can you please add some docs for:
```
// TODO: some docs
FromContains<Equals<ParentLocation>, pallet_ah_migrator::ExceptResponseFor<Equals<Here>>>,
```
Could you please add a test covering this? Is `Equals<Here>` the correct value? Does `Here` applies from the `ParentLocation` point of view?

Where do we even send `QueryResponse`? Is it because the `xcm-executor` can send it internally? Can I see somewhere the flow which triggers `QueryResponse`?

I'm trying to read the `to_querier` function, the `QueryResponse` instruction, and the surrounding code, but it's not exactly easy bedtime reading :)